hash_practice/exercises.py

```diff
@@ -1,19 +1,50 @@
 
-def grouped_anagrams(strings):
+def grouped_anagrams(words):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)
+        Space Complexity: O(n)
     """
-    pass
+
+    array = []
+    dic = {}
+    for word in words:
+        char_in_words_sum = 0
+        for char in word:
+            char_in_words_sum += ord(char)
+        if char_in_words_sum in dic:

```

Comment:
So this is an interesting way to generate a key here, but there's a bug. I'm disappointed our automated tests didn't catch it!

In short, you can come up with multiple different strings that may not have the same letters at all but do have the same sum. The simplest counter-example I could come up with is "dd" and "ce". `ord("d")` is 100, `ord("c")` is 99, and `ord("e")` is 101. In this case the sum for both "dd" and "ce" is 200, but they are not anagrams of each other.

Like I said, as it passed our tests, but this bug drives me to mark this down to Yellow.